api/envoy/config/core/v3/socket_interfaces.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.config.core.v3;
+
+import "envoy/config/core/v3/extension.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.core.v3";
+option java_outer_classname = "SocketInterfacesProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Socket Interfaces Configuration ]
+
+// Used to select the socket interface implementation to use
+message SocketInterfacesConfig {
+  // Addition socket interfaces to initialize
+  repeated TypedExtensionConfig custom_types = 1;
+
+  // Override default socket interface
+  TypedExtensionConfig custom_default_type = 2;
```

Comment:
Should this just be a string? I.e. the extension config is above, this is just specifying some entry in it.

Reply:
Done!